workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSReceiveSystem.cs

```diff
@@ -77,15 +78,13 @@ protected override void OnUpdate()
             while (inCriticalSection);
         }
 
-        private void OnAddEntity(AddEntityOp op)
+        internal void OnAddEntity(AddEntityOp op)
         {
             var entityId = op.EntityId;
             if (worker.EntityIdToEntity.ContainsKey(entityId))
             {
-                worker.LogDispatcher.HandleLog(LogType.Error, new LogEvent(Errors.DuplicateAdditionOfEntity)
-                    .WithField(LoggingUtils.LoggerName, LoggerName)
-                    .WithField(LoggingUtils.EntityId, entityId));
-                return;
+                throw new InvalidSpatialEntityStateException(
```

Comment:
Not a huge fan of this setup - you can't easily ensure that you're interpolating the correct arguments.

Reply:
Hmm - one thing I could do is pull the interpolation into a helper function so that function has named arguments